soda/bigquery/soda/data_sources/bigquery_data_source.py

```diff
@@ -153,6 +153,12 @@ def sql_get_table_names_with_count(
         where_clause = f"\nWHERE {table_filter_expression} \n" if table_filter_expression else ""
         return f"SELECT table_id, row_count \n" f"FROM {self.dataset_name}.__TABLES__" f"{where_clause}"
 
+    def sql_get_table(self, table_name: str, limit: int = None) -> str:
+        sql = f"SELECT * \n" f"FROM {table_name}"
+        if limit is not None:
+            sql += f" \n LIMIT {limit}"
```

Comment:
I have an allergy for string concatenation like this as it makes the code harder to maintain.

I would advise to have the template at the end, and have your ifs either generate an empty str or not before with explicit names for each part. This means, someone who would later add anything to the code doesn't need to make sure they place their code in exactly the right order.

In this instance, I would:
- put the SQL string at the end as an `f` string with all placeholders in the right order
- store each part of the template before with `''` when they don't get executed.

Hope this is clear if not, I can just show you a concrete example of what I mean.

Reply:
good point i ll try to keep that in mind as well